File: src/DesignPattern/Creational/SimpleFactory/SimpleFactory-go/SimpleFactory.go
package simplefactory

import "fmt"

type product interface {
	getColor()
}

// 设计模式：创建型-简单工厂
// 算法：——
// 复杂度：——
func productFactory(arg string) product {
	switch arg {
	case "A":
		return &productA{}
	case "B":
		return &productB{}
	}
	return nil
}

type productA struct{}

func (p *productA) getColor() {
	fmt.Println("red")
}

type productB struct{}

func (p *productB) getColor() {
	fmt.Println("black")
}

File: src/DesignPattern/Creational/SimpleFactory/SimpleFactory-java/simpleFactory.java

interface product {
    void getColor();
}

class productA implements product {
    public void getColor() {
        System.out.println("red");
    }
}

class productB implements product {
    public void getColor() {
        System.out.println("blank");
    }
}



public class simpleFactory {
    enum types {
        A, B,
    };

    public static product createFactory(types t) {
        switch (t) {
        case A:
            return new productA();
        case B:
            return new productB();
        }
        return null;
    }
}